### PR TITLE
tests: cdrouter_bootstrap: modified cdrouter provision setup

### PR DESCRIPTION
**Pull request by MartinCBN**
modified cdrouter provison setup
add recover dut to CM_online_freq

Signed-off-by: MartinCBN <martin_wu@compalbn.com>

### CODE REVIEW

File: tests/cdrouter_bootstrap.py [hunk 1]
```diff
@@ -323,6 +334,15 @@ def recover(self):
...
             board.wait_for_linux()
 
+        wan_ip = board.get_interface_ipaddr(board.wan_iface)
+        cm_freq = self.config.board['common']['CM_online_freq']
+        wan.sendline("snmpset -v 2c -c public -t 30 -r 3 "+wan_ip+" "+board.mib["lastKnownFreq"]+".0 i "+cm_freq)
+        assert 0 != wan.expect(['Timeout: No Response from'] + prompt)
+        board.enter_arm_shell()
+        board.arm.sendline('reboot')
+        board.wait_for_linux()
+        board.wait_for_network()
```
Comment (mattsm) [inline tests/cdrouter_bootstrap.py]: This is not generic, board specific stuff can't go here....

File: tests/cdrouter_bootstrap.py [hunk 2]
```diff
@@ -58,6 +58,16 @@ def runTest(self):
 
         c = CDRouter(self.cdrouter_server)
 
+        """Go to cdrouter frequency and reboot to Dual-stack config"""
+        wan_ip = board.get_interface_ipaddr("wan0")
+        if self.config.board['cdrouter_location'] == 'CBN':
```
Comment (mattsm) [inline tests/cdrouter_bootstrap.py]: I don't think using the location is the right thing to flag off of here.

File: tests/cdrouter_bootstrap.py [hunk 3]
```diff
@@ -58,6 +58,16 @@ def runTest(self):
...
+            cm_freq = self.config.board['cdrouter_freq']
+        else:
+            cm_freq = self.config.board['common']['CM_online_freq']
```
Comment (mattsm) [inline tests/cdrouter_bootstrap.py]: I thought we talked with @SyHuangCBN about this? Can't we drop this freq from the config and determine it dynamcially? (From a mac domain listed in the board config?)

File: tests/cdrouter_bootstrap.py [hunk 4]
```diff
@@ -58,6 +58,16 @@ def runTest(self):
...
+        wan.sendline("snmpset -v 2c -c public -t 30 -r 3 "+wan_ip+" "+board.mib["lastKnownFreq"]+".0 i "+cm_freq)
+        assert 0 != wan.expect(['Timeout: No Response from'] + prompt)
+        board.reprovision(provisioner, cm_cfg="9_EU_CBN_Dual-Stack_LG.txt", mta_cfg="AutoCA.txt")
```
Comment (mattsm) [inline tests/cdrouter_bootstrap.py]: This is all board specific... it need to be generic. If you need to do something board specific pre and post run you can add a hook... or possibly just having a test that inherits and runs the provisioning code before starting CDrouter...
```
class myCDrouterTest(CDrouterBootStrap):
   def runTest(self):
       board.provision()
       super(runTest())
```

File: tests/cdrouter_bootstrap.py [hunk 5]
```diff
@@ -176,21 +186,22 @@ def add_cdrouter_config(config):
...
+    testvar wanIspAssignIp %s
+    testvar wanNatIp %s
+    testvar IPv4HopCount %s
+    testvar lanDnsServer %s
+    testvar wanDnsServer %s""" % (self.config.board['cdrouter_wanispip'], \
+                                  self.config.board['cdrouter_wanispgateway'], \
+                                  wan_ip, wan_ip, \
+                                  self.config.board['cdrouter_ipv4hopcount'], \
+                                  board.get_dns_server(), \
+                                  board.get_dns_server_upstream())
```
Comment (mattsm) [inline tests/cdrouter_bootstrap.py]: These need to be defined in the boardfarm config for CBN... not just overridden. They should also be removed from configs/cdrouter-mv1.conf only generic items should live in that config.

**Comment by mattsm** [comment]
Any followup- can't add board specific items here.

File: tests/cdrouter_bootstrap.py (hunk 4 above)
Comment (MartinCBN) [reply tests/cdrouter_bootstrap.py]: Understood, running the provisioning code before testing cdrouter.

File: tests/cdrouter_bootstrap.py (hunk 2 above)
Comment (MartinCBN) [reply tests/cdrouter_bootstrap.py]: This location is used for testing cdrouter with another way. The NTL team(maual test team) using cdrouter to go to specific frequency to test case, not we usually using frequency. Therefore, I just using this location to change frequency. Is it proper?

File: tests/cdrouter_bootstrap.py (hunk 3 above)
Comment (MartinCBN) [reply tests/cdrouter_bootstrap.py]: Okay, I will modified this part after mac domain listed in the board config

**Comment by mattsm** [comment]
This is old, going to close... need to move cdrouter to device model to take changes like this...

**Pull request closed without merging**